src/pretix/presale/views/order.py

```diff
@@ -230,8 +236,15 @@ def download_buttons(self):
     def get_context_data(self, **kwargs):
         ctx = super().get_context_data(**kwargs)
         ctx['cart'] = self.get_cart(
-            answers=True, downloads=ctx['can_download'],
-            queryset=self.order.positions.prefetch_related('issued_gift_cards').select_related('tax_rule'),
+            answers=True,
+            downloads=ctx['can_download'],
+            queryset=self.order.positions.prefetch_related('issued_gift_cards').select_related('tax_rule').annotate(
+                checkin_count=Subquery(
+                    Checkin.objects.filter(
+                        successful=True, type=Checkin.TYPE_ENTRY, position_id=OuterRef('pk')
+                    ).order_by().values('position').annotate(c=Count('*')).values('c')
+                )
+            ),
```

Comment:
See my comment above regarding annotating checkin_count only if `show_checkin_number_user` is set.